src/vmm/src/config.rs

```diff
@@ -79,26 +172,12 @@ impl TryFrom<String> for VcpuConfig {
     type Error = ConversionError;
 
     fn try_from(vcpu_cfg_str: String) -> result::Result<Self, Self::Error> {
-        // Supported options: `num_vcpus=<u8>`
-        let vcpu_cfg_str_lower = vcpu_cfg_str.to_lowercase();
-        let tokens: Vec<&str> = vcpu_cfg_str_lower
-            .split('=')
-            .filter(|tok| !tok.is_empty())
-            .collect();
-        if tokens.len() != 2 {
-            return Err(ConversionError::ParseVcpus(vcpu_cfg_str));
-        }
-        if tokens[0] != "num" {
-            return Err(ConversionError::ParseVcpus(tokens[0].to_string()));
-        }
-        let vcpu_config = tokens[1]
-            .parse::<u8>()
-            .map(|num_vcpus| VcpuConfig { num: num_vcpus })
-            .map_err(|_| ConversionError::ParseVcpus(tokens[1].to_string()))?;
-        if vcpu_config.num == 0 {
-            return Err(ConversionError::ParseVcpus(tokens[1].to_string()));
-        }
-        Ok(vcpu_config)
+        // Supported options: `num=<u8>`
+        let mut tokens = tokenize(vcpu_cfg_str);
+        let num = parse_arg(&mut tokens, "num", Some(num::NonZeroU8::new(1).unwrap()))
+            .map_err(ConversionError::new_vcpus)?;
+        check_all_consumed(&tokens).map_err(ConversionError::new_vcpus)?;
```

Comment:
I think one can get rid of calling this function completely. 

At the end of all the calls to `parse_arg` calls. Just check the `tokens` is empty and if not create an error and return. That's perhaps still some boiler plate, but won''t require to make this function call.

Reply:
I'm not sure if I got what you mean, this function call is exactly doing that. I think it's cleaner to have the function call for this check and create the related error there (once for every TryFrom functions). If we ever need to update the error format, one would just need to update it in that function, once. Did I address correctly your comment?

Reply:
Yes, you are right but something that can be done with a simple `if` / `else`. But I am not overly opinionated about it, the function call is just fine too. In fact - in general - unprocessed arguments should ideally be simply `log::warn` and need not be a failure as they are not doing anything wrong or unpleasantly surprising (and this is why I simply prefer `if`/`else` - in fact only `if`). But the way it is - is just fine too.

Reply:
It was just my impression that the parsing should fail if the tokens are not all consumed - I was thinking about mistypings - but I think you're right, maybe a `log::warn` could be enough. @andreeaflorescu can you provide some input on this?